crates/web-sys/webidls/unstable/FileSystemAccess.webidl

```diff
@@ -0,0 +1,72 @@
+enum FileSystemPermissionMode {
+  "read",
+  "readwrite"
+};
+
+dictionary FileSystemPermissionDescriptor : PermissionDescriptor {
+  required FileSystemHandle handle;
+  FileSystemPermissionMode mode = "read";
+};
+
+dictionary FileSystemHandlePermissionDescriptor {
+  FileSystemPermissionMode mode = "read";
+};
+
+[Exposed=(Window,Worker), SecureContext, Serializable]
+partial interface FileSystemHandle {
+  [Throws]
+  Promise<PermissionState> queryPermission(optional FileSystemHandlePermissionDescriptor descriptor = {});
+  [Throws]
+  Promise<PermissionState> requestPermission(optional FileSystemHandlePermissionDescriptor descriptor = {});
```

Comment:
```suggestion
  Promise<PermissionState> queryPermission(optional FileSystemHandlePermissionDescriptor descriptor = {});
  Promise<PermissionState> requestPermission(optional FileSystemHandlePermissionDescriptor descriptor = {});
```
Looking at the spec I can#t see that these methods throw, their `Promise` might be rejected, but that's different, adding `[Throws]` do a method that returns a `Promise` will guard the method and not the `Promise`.

For potentially rejected `Promise`s nothing has to be done, they are guarded and return `Result` by default.